frontend/src/components/Accordion.vue

```diff
@@ -1,50 +1,67 @@
+                    {{ list.name }}
+                </button>
+            </h3>
 
-    <div :id="collapseId(list.id)" class="accordion-collapse collapse" :aria-labelledby="`#${headingId(list.id)}`" :data-bs-parent="`#${accName}`">
-      <div class="accordion-body">
+            <div :id="collapseId(index)" class="accordion-collapse collapse" :aria-labelledby="`#${headingId(index)}`" :data-bs-parent="`#${accName}`">
+                <div class="accordion-body">
+                    <p v-if="list.description">{{ list.description}}</p>
+                    // @todo make different components
+                    <template v-if="list.type === 'category'">
+                        <div class="form-check" v-for="item in categoryItems(list)"
+                             :key="item.id">
+                            <input class="form-check-input" type="radio" value="" :name="list.id" :id="item.id" @change="updateEntityValue(list.name, $event.target.value, list.type)">
+                            <label class="form-check-label" :for="item.id">
+                                {{item.name}}
+                            </label>
+                        </div>
+                    </template>
+                    <template v-else>
+                        // @todo there will be special fields, like picking a location, nationality etc
+                        <label :for="list.name" class="form-label">{{ list.name }}</label>
+                        <input type="" class="form-control" :id="list.id" :value="entityValue(list.name, list.type)" @change="updateEntityValue(list.name, $event.target.value, list.type)" placeholder="">
+                    </template>
 
-        <p>{{ list.description}}</p>
-        <div class="form-check" v-for="item in codelistItems(list.id)"
-             :key="item.id">
-          <input class="form-check-input" type="radio" value="" :name="list.id" :id="item.id" @change="registerInputChange(list, item)">
-          <label class="form-check-label" :for="item.id">
-            {{item.name}}
-          </label>
+                    <slot></slot>
+                </div>
+            </div>
         </div>
-
-
-        <slot></slot>
-
-
-      </div>
     </div>
-  </div>
-</div>
 </template>
 <script>
-export default {
-  name: "Accordion",
-  props: ['lists', 'accName'],
-  methods: {
-    codelistItems(listId) {
-      return this.$store.getters.getCodelistItemsByListId(listId)
-    },
-    headingId(id) {
-      return `heading-${id}`
-    },
-    collapseId(id) {
-      return `collapse-${id}`
-    },
-    registerInputChange(list, item) {
-      this.$emit('inputChange', list, item)
-    }
-  }
-}
+ import { mapState, mapGetters, mapActions } from 'vuex'
+
+ export default {
+     name: "Accordion",
+     props: ['lists', 'accName'],
+     methods: {
+         ...mapActions(['updateCurrentRecordProperty']),
+         entityValue(name, type) {
+             return this.getEntityValue(this.title, name, type)
```

Comment:
The title variable is not defined in this component.

```suggestion
             return this.getEntityValue(this.$route.params.entityname, name, type)
```